PCDtoPointCloud2Publisher:
  ros__parameters:
    # 输入路径（单文件或文件夹）
    inputPath: "data/PCD/scans_1.pcd"
    # 是否启用文件夹模式（true为文件夹，false为单文件）
    inputFolderMode: false
    FolderOptions: # 文件夹模式参数（仅当inputFolderMode为true时生效）
      # 播放模式（可选值：sequential：按照文件夹顺序播放, parallel：播放全部文件夹, split：将文件夹视为分包）
      playMode: "sequential"
      # 分包播放参数（仅当playMode为split时生效）
      splitFilePrefix: "scans_"        # 分包文件名的前缀（如"scan_"）
      splitStartIndex: 0               # 分包起始编号（如从"scan_0000.pcd"开始）
    # 话题命名配置
    # ROS命名空间（组织话题层级，如 "/sensors/lidar"）
    topicNamespace: "point_cloud"
    # 主话题名（与命名空间组合，如 topicNamespace + "/" + topicName）
    topicName: "data"                  # 默认话题为 "point_cloud/data_0"
    # 坐标系ID（如激光雷达坐标系）
    frameId: "map"
    # 高级选项
    # 是否启用多线程加载下一个文件（减少延迟）
    enableMultiThreadLoad: false
    # 时间戳模式（"system"或"file"，file需PCD包含时间戳）
    timestampSource: "system"
    # 时间戳增量（单位：秒，用于模拟传感器时间）
    timestampIncrementSeconds: 0.0
    # 日志选项
    # 是否启用详细日志输出
    enableDebugLogging: true
    # 日志文件路径（留空则不保存）
    logFilePath: ""
    # PCL点类型（需与PCD文件匹配，如"PointXYZ"或"PointXYZI"）
    pclPointType: "PointXYZI"
    # 公共播放参数
    # 播放速率（>1快放，<1 慢放）
    playRate: 1.0
    # 是否循环播放（单文件模式循环文件，文件夹模式循环整个序列）
    playLoop: false